irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -1645,15 +1664,93 @@ private static Inode toFh(long _inodeNumber)
+        log_.debug("""
+                vfs::tryAndFindPathFromGenQuery
+                tryAndFindPathFromGenQuery - _inodeNumber = {}
+                tryAndFindPathFromGenQuery - option       = {}""", 
+                _inodeNumber, option);
+
+        String strPath = null;
+        IRODSGenQueryBuilder builder;
+        IRODSGenQueryExecutor gqe;
+        IRODSGenQueryFromBuilder query;
+        IRODSQueryResultSet resultSet;
+
+
+        if (option.equals("data")) {
+            try {
+                builder = new IRODSGenQueryBuilder(true, null);
+    
+                builder.addSelectAsGenQueryValue(RodsGenQueryEnum.COL_COLL_ID)
+                    .addSelectAsGenQueryValue(RodsGenQueryEnum.COL_COLL_NAME)
+                    .addConditionAsGenQueryField(RodsGenQueryEnum.COL_COLL_ID, QueryConditionOperators.EQUAL, String.valueOf(_inodeNumber));
+    
+                gqe = factory_.getIRODSGenQueryExecutor(adminAcct_);
+                query = builder.exportIRODSQueryFromBuilder(50);
+                resultSet = gqe.executeIRODSQueryAndCloseResultInZone(query, 0, adminAcct_.getZone());
+    
+                log_.debug("found row is (coll id) = {}", resultSet.getResults());
+                if (resultSet.getResults().size() != 0) {
+                    strPath = resultSet.getResults().get(0).getColumn(1);   
+                    log_.debug("path from coll_id = {}", strPath);
+                }
+    
+                                        
+            } catch (GenQueryBuilderException | JargonException | JargonQueryException e)
+            {
+                log_.error(e.getMessage());
+            }    
+        } else if (option.equals("coll")) {
+            try {
+                builder = new IRODSGenQueryBuilder(true, null);
+    
+                builder.addSelectAsGenQueryValue(RodsGenQueryEnum.COL_D_DATA_ID)
+                    .addSelectAsGenQueryValue(RodsGenQueryEnum.COL_COLL_NAME)
+                    .addSelectAsGenQueryValue(RodsGenQueryEnum.COL_DATA_NAME)
+                    .addConditionAsGenQueryField(RodsGenQueryEnum.COL_D_DATA_ID, QueryConditionOperators.EQUAL, String.valueOf(_inodeNumber));
+    
+                gqe = factory_.getIRODSGenQueryExecutor(adminAcct_);
+                query = builder.exportIRODSQueryFromBuilder(50);
+                resultSet = gqe.executeIRODSQueryAndCloseResultInZone(query, 0, adminAcct_.getZone());
+    
+                log_.debug("found row is (data id) = {}", resultSet.getResults());
+                if (resultSet.getResults().size() != 0) {
+                    strPath = resultSet.getResults().get(0).getColumn(1) + '/' + resultSet.getResults().get(0).getColumn(2);   
+                    log_.debug("combined path is = {}", strPath);
+                }                       
+            } catch (GenQueryBuilderException | JargonException | JargonQueryException e)
+            {
+                log_.error(e.getMessage());
+            }
+    
+        }
```

Comment:
do we need a fallthrough else case here?

Reply:
I don't think we _need_ one, since the default value of `strPath` will be null. However, I think that printing a debug message would help. Something like log_.warn("option [{}] is not supported", option);

happy to discuss

Reply:
yes, always nice to have a 'something has gone wrong - shouldn't be here...' type of message.

Reply:
Sweet, it's in the latest commit!